api/src/main/java/org/openmrs/module/oauth2login/authscheme/UserInfo.java

```diff
@@ -92,20 +92,17 @@ public String toString() {
 	 * @return The user info JSON value for the specified OpenMRS property key.
 	 */
 	public Object get(String propertyKey) throws RuntimeException {
-		if (!props.containsKey(propertyKey)) {
-			throw new IllegalArgumentException("There was an attempt to read the corresponding value for '" + propertyKey
-			        + "' in the user info JSON, this property is not mapped in the OAuth2 properties file.");
+		if (props.containsKey(propertyKey)) {
+			String jsonKey = props.getProperty(propertyKey);
+			try {
+				return JsonPath.read(json, "$." + jsonKey);
+			}
+			catch (PathNotFoundException e) {
+				throw new PathNotFoundException("There was an error when reading the JSON path $." + jsonKey
+				        + " mapped from '" + propertyKey + "' in the user info JSON.", e);
+			}
 		}
-		
-		String jsonKey = props.getProperty(propertyKey);
-		try {
-			return JsonPath.read(json, "$." + jsonKey);
-		}
-		catch (PathNotFoundException e) {
-			throw new PathNotFoundException("There was an error when reading the JSON path $." + jsonKey + " mapped from '"
-			        + propertyKey + "' in the user info JSON.", e);
-		}
-		
+		return null;
```

Comment:
@icrc-jfigueiredo basically what you have done here is that unmapped props will return `null`, that's fine. I would say, stick to that rule and do null checks every time you need to set something when updating a `User`.

Reply:
@icrc-jfigueiredo this is not a null check:
```java
StringUtils.isNotEmpty(userInfo.getOpenmrsUser().getPersonName().getGivenName())
```
You should eventually end up with something like:
```java
userInfo.getOpenmrsUser().getPersonName().getGivenName() != null
```
Same logic for anything else coming out of `UserInfo`, incl. the list of role names and the gender.

Reply:
@mks-d actually, that method does something like this : 
`    public static boolean isEmpty(String str) {
        return str == null || str.length() == 0;
    }

   `